### PR TITLE
Update matter IDL to contain per-endpoint attribute storage and defaults

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Need to codereview storage data. Specifically seeing featuremap defaults and cluster revisions in a reviewable manner is very useful.

#### Change overview
Update endpoint information to contain storage and defaults as a section. Updated docs, unit tests and parsers. They look like:

```
// ....

endpoint 1 {
   // ...

   server cluster WindowCovering {
    ram      attribute type;
    persist  attribute currentPositionLift;
    persist  attribute currentPositionTilt;
    persist  attribute configStatus default = 0x03;
    persist  attribute currentPositionLiftPercentage;
    persist  attribute currentPositionTiltPercentage;
    ram      attribute operationalStatus;
    ram      attribute targetPositionLiftPercent100ths;
    ram      attribute targetPositionTiltPercent100ths;
    ram      attribute endProductType;
    persist  attribute currentPositionLiftPercent100ths;
    persist  attribute currentPositionTiltPercent100ths;
    persist  attribute installedOpenLimitLift;
    persist  attribute installedClosedLimitLift default = 0xFFFF;
    persist  attribute installedOpenLimitTilt;
    persist  attribute installedClosedLimitTilt default = 0xFFFF;
    persist  attribute mode;
    ram      attribute safetyStatus;
    ram      attribute featureMap default = 0x0017;
    ram      attribute clusterRevision default = 5;
  }
}

```

#### Testing
- Unit tests
- Visual inspection of output
- Built Android chip-tool (which parses matter idl files for codegen)
- CI will validate as well

### CODE REVIEW

**Comment by andy31415** [comment]
Note: this depends on #18046 (includes the latest zap pull)

File: examples/bridge-app/bridge-common/bridge-app.matter [hunk 1]
```diff
@@ -1134,31 +1134,289 @@ server cluster WiFiNetworkDiagnostics = 54 {
...
+    callback attribute hardwareVersionString;
+    callback attribute softwareVersion;
+    callback attribute softwareVersionString;
+    callback attribute manufacturingDate default = "20210614123456ZZ";
+    callback attribute partNumber;
+    callback attribute productURL;
+    callback attribute productLabel;
+    callback attribute serialNumber;
+    persist  attribute localConfigDisabled;
+    ram      attribute reachable default = 1;
```
Comment (bzbarsky-apple) [inline examples/bridge-app/bridge-common/bridge-app.matter:1174]: Really, we should disable this attribute.  It's not really meant to be used on this cluster....  Followup for that?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/bridge-app/bridge-common/bridge-app.matter (hunk 1 above)
Comment (andy31415) [reply examples/bridge-app/bridge-common/bridge-app.matter:1174]: Created #18083 

Us being able to see this more directly (in theory this is also somewhat viewable in cluster_endpoints.h) is one of the goals of having IDLs.

File: scripts/idl/test_matter_idl_parser.py [hunk 2]
```diff
@@ -380,9 +344,9 @@ def test_cluster_event_acl(self):
 
     def test_multiple_clusters(self):
         actual = parseText("""
-            server cluster A = 1 {}
-            client cluster B = 2 {}
-            client cluster C = 3 {}
+            server cluster A = 1 { /* FIXME */ }
```
Comment (tehampson) [inline scripts/idl/test_matter_idl_parser.py:347]: Were you intending to do something here?

Comment (andy31415) [reply scripts/idl/test_matter_idl_parser.py:347]: oops ... I think this was a reminder. I will rename it to test-comment.

**Review by tehampson — APPROVED** [review]

**Pull request merged**